docs/revsets.md

```diff
@@ -70,6 +70,38 @@ only symbols.
 You can use parentheses to control evaluation order, such as `(x & y) | z` or
 `x & (y | z)`.
 
+### The `all:` prefix
```

Comment:
nit: I would move this section around "Aliases" or "Examples". As you mentioned, it's not technically a language construct, and can't be inserted in the middle of expressions.

Reply:
I'm unsure what's best, though either way seems OK. It *looks*  like a language construct but isn't.

The interaction that caused this PR was somebody asking "what is the `:` operator when they saw `jj rebase -s all:...` suggested in a forum. For this situation, placing it next to the language construct seems better. On the other hand, perhaps we shouldn't over-optimize for this one anecdote.

Reply:
I moved the section down as you requested. It should be fine. If enough people seem to be unable to find it when they need it, I might be tempted to move it up again.